### PR TITLE
Add MultipleFileField to handle and clean multiple files

### LINKED ISSUE #464: Incompatibility with latest Django 3.2.19 security release
See https://www.djangoproject.com/weblog/2023/may/03/security-releases/

Django 3.2.19 disabled the ability to upload multiple files using one form field (see https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965). Unfortunately, we started consuming this mechanism in https://github.com/ome/omero-web/pull/410 to allow multiple file annotations to be uploaded  via the UI.

Currently if Django is installed or upgraded to 3.2.19, OMERO.web 5.17+ will fail to start with the following exception

```
ValueError: ClearableFileInput doesn't support uploading multiple files.
```

As an immediate workaround, capping Django to 3.2.18 should restore the ability to start OMERO.web. This issue should likely be addressed properly in a patch release of OMERO.web restoring compatibility with the Django 3.2.x line.

My inclination would be to disable the feature introduced in #410 for now and look into reintroducing it at a later date using a supported API. 

/cc @knabar @chris-allan @will-moore @pwalczysko @jburel

### PR DESCRIPTION
**Pull request by will-moore**
Fixes #464

This adds the update suggested at the last link below to conform to the security release below:

https://github.com/ome/omero-web/pull/410
https://www.djangoproject.com/weblog/2023/may/03/security-releases/
https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965

To test:
 - Build should be green (using latest Django 3.2.19)
 - Multiple file upload should still be working
 - When uploading multiple files to file annotations, ALL files should be validated (e.g. to prevent empty files to be uploaded) instead of only the last file being validated.

### CODE REVIEW

**Review by sbesson — CHANGES_REQUESTED** [review]
See https://github.com/ome/omero-web/pull/466 which exposes another underlying issue related to the validation of the files uploaded via the `FilesAnnotationForm` (and allows to create a scenario to test this PR).
While testing the two PRs together, it should be noted that this PR _requires_ Django 3.2.19 as the API for auto-setting  `multiple: true` when `allow_multiple_selected` is set to True was introduced in https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965. In other terms, with this PR and Django 3.2.18, the multiple selection is disabled. Django should be set to 3.2.19 as the minimum version via `setup.py`

**Comment by will-moore** [comment]
Excluding to allow testing of #466 ahead of this...

--exclude

**Comment by will-moore** [comment]
Testing on merge-ci:

 - upload 1 or more non-empty files works -> Creates FileAnnoations
 - upload multiple Files which includes an empty file first, last or middle (if 3) fails silently - (form is invalid).

File: omeroweb/webclient/forms.py [hunk 1]
```diff
@@ -62,6 +62,28 @@
 ) % help_button
 
 
+#################################################################
+# Custom widget and validation for multiple file uploads
```
Comment (sbesson) [inline omeroweb/webclient/forms.py:66]: Having looked at how the code is constructed, it looks like the approach is do declare all custom widget/fields in `omeroweb.webclient.custom_forms` and import them here to construct the forms.

File: omeroweb/webclient/forms.py [hunk 2]
```diff
@@ -62,6 +62,28 @@
...
+
+
+class MultipleFileInput(forms.ClearableFileInput):
+    allow_multiple_selected = True
```
Comment (sbesson) [inline omeroweb/webclient/forms.py:70]: Might be worth adding a comment and/or a reference to the documentation explaining why we are subclassing these Django classes.

**Review by sbesson — CHANGES_REQUESTED** [review]

**Comment by dominikl** [comment]
Functional test looks good 👍  If there's one empty file included in a multi file upload then none of the files get uploaded/attached, not even the non-empty ones. Is that expected?

**Comment by sbesson** [comment]
@dominikl that's the expectation as per the new field class. If any of the files fails validation, then the entire multi-file upload is cancelled. 


The other place that uses a `FileField` is the photo upload in `My account` and this shows an informative message if an empty file is uploaded 

Screenshot 2023-05-05 at 12 58 02

@will-moore is there a way to propagate the error message to the user in the files annotation UI?

**Comment by dominikl** [comment]
👍 I don't think either that it has to go into this PR, better get it out as soon as possible.

**Review by sbesson — APPROVED** [review]
The last commit follows the new recommendation from Django documentation with a new custom class and restores compatibility with Django 3.2.19.

The last commit will need to be retested functionally but from a code review perspective, no objection to moving forward with the current suggestion.

**Review by knabar — APPROVED** [review]
Code looks good to me

**Comment by will-moore** [comment]
Running https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/1446/ to include latest changes... 
Built with https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-build/1402/
and deployed with https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-web/1659/

**Comment by pwalczysko** [comment]
Tested on merge-ci, user-3. From jenkins, this PR is included on https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/1447/console


```
Merged PRs:
  - PR 425 will-moore 'Add hook for mapr or other app to extend webclient main page'
  - PR 456 kkoz 'Allow Tiled Image Histograms'
  - PR 462 will-moore 'OME.formatDate handles null input'
  - PR 465 will-moore 'Add MultipleFileField to handle and clean multiple files'
  - PR 467 knabar 'Add change log for 5.20.0'
  - PR 469 will-moore 'Temp setGroupForSession to edit current group'
 ```

The test:
1. Create an empty file using ``touch`` on the cli and try to attach it in the Attachments harmonica of webclient. 
2. Try to attach the empty file also with other, non-empty files
3. Try to upload the empty file as a photo in user settings of web UI

The tests went as expected, the file upload was rejected in all cases where the empty file was selected, either with non-empty files or alone.
The upload of photos and files (non-empty ones) still works.

LGTM

**Review by pwalczysko — APPROVED** [review]

**Review by jburel — APPROVED** [review]

**Pull request merged**